vault_core/src/vault.rs

```diff
@@ -794,9 +797,23 @@ impl Vault {
             return Err(VaultError::VaultInsufficientFunds);
         }
 
-        let fee_amount = self.calculate_withdraw_fee(amount_in)?;
+        let program_fee_amount = Config::calculate_program_fee(program_fee_bps, amount_in)?;
+        let mut vault_fee_amount = self.calculate_withdraw_fee(amount_in)?;
+
+        // Prioritize program fee over vault fee if they exceed the amount in
+        if program_fee_amount
```

Comment:
this seems wrong?

Reply:
thought we had decided that we will ensure program fee gets its full share, and then vault fee will get whatever is left if they sum above 100 - is that not desired? or is the implementation here wrong?

Reply:
actually this can just be simplified to: let fee = min(amount_in, program_fee_amount + vault_fee_amount)?

Reply:
this gets us the total fee amount but we'd still need logic to adjust vault fee if needed right?